### ISSUE TITLE
Creating identical Authentication Executions in multiple realms not possible anymore after version upgrade

### ISSUE THREAD

**Issue by arberselmani** [keycloak/keycloak#21434]
### Before reporting an issue

- [X] I have searched existing issues
- [x] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

admin/api

### Describe the bug

After upgrading from v21.0.1 to v21.1.1, I am dealing with a problem when trying to create Authentication Executions within an Authentication Flow via Rest API.
It is not possible to create authentication executions in multiple realms with the same authenticator script.

### Version

21.1.1

### Expected behavior

An Authentication Execution with the same authenticator script can be created and used in multiple realms, not only once for all realms. Also, after deleting an execution, it should be possible to create it again.

### Actual behavior

When trying to create the Authentication Execution with the same authenticator script in multiple realms, it will be applied only to the first realm. After that, Keycloak throws an error.
Same happens even when deleting the execution and then trying to add it again. 

### How to Reproduce?

I am using a custom authenticator script for that, which is already provided in `META_INF/keycloak-scripts.json`:
```
{
  "authenticators": [
    {
      "name": "shared-script",
      "fileName": "shared/script.auth.js",
      "description": ""
    }
  ]
}
```

Then I call the POST endpoint `/admin/realms/{realm}/authentication/flows/{flow_alias}/executions/execution` with the body:
```
{
  "provider": "shared/script.auth.js"
}
```

Now the trouble begins. Since this process is needed for multiple realms, they all use the same authenticator script.
The problem that I am facing now is that it works only the first time. As soon as I try to apply it to other realms, I get the following error:

```
keycloak-keycloak-1  | 2023-07-04 11:50:24,301 ERROR [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-9) 
Unique index or primary key violation: "PUBLIC.PRIMARY_KEY_C1 ON PUBLIC.AUTHENTICATOR_CONFIG(ID) VALUES ( /* 45 */ 'shared/script.auth.js' )"; SQL statement:
keycloak-keycloak-1  | insert into AUTHENTICATOR_CONFIG (ALIAS, REALM_ID, ID) values (?, ?, ?) [23505-214]
keycloak-keycloak-1  | 2023-07-04 11:50:24,302 INFO  [org.hibernate.engine.jdbc.batch.internal.AbstractBatchImpl] (executor-thread-9) HHH000010: On release of batch it still contained JDBC statements
```

The same error occurs even if I delete the executions via Admin Console and try to add it manually. They still seem to be somewhere in the DB which causes this violation.

**Comment by pedroigor** [comment]
@arberselmani It might be fixed by https://github.com/keycloak/keycloak/issues/12012.

Could you try the nightly container image or build from upstream/main sources?

**Comment by chrisFrodo** [comment]
> @arberselmani It might be fixed by #12012.
> 
> Could you try the nightly container image or build from upstream/main sources?

Hi !
Check my comment here : https://github.com/keycloak/keycloak/issues/20005#issuecomment-1619641595

Thanks.

**Comment by arberselmani** [comment]
> @arberselmani It might be fixed by #12012.
> 
> Could you try the nightly container image or build from upstream/main sources?

Thanks for the reply!

I tested it with the nightly container image and unfortunately I get the same error.

```
keycloak-keycloak-1  | 2023-07-18 07:13:00,523 ERROR [org.hibernate.engine.jdbc.spi.SqlExceptionHelper] (executor-thread-3) 
Unique index or primary key violation: "PUBLIC.PRIMARY_KEY_C1 ON PUBLIC.AUTHENTICATOR_CONFIG(ID) VALUES ( /* 35 */ 'shared/script.auth.js' )"; SQL statement:
keycloak-keycloak-1  | insert into AUTHENTICATOR_CONFIG (ALIAS,REALM_ID,ID) values (?,?,?) [23505-220]
```

**Comment by mposolda** [comment]
duplicates #20005 

@arberselmani Thanks for the report, but I am closing as it is duplicate of another issue. Please follow the other issue for the updates regarding this.
